simulator/src/main/cook/sim/travis.clj

```diff
@@ -0,0 +1,78 @@
+(ns cook.sim.travis
+  (:require [clojure.set :refer [intersection]]
+            [clojure.string :as string]
+            [clojure.tools.cli :refer [parse-opts]]
+            [clj-http.client :as http]
+            [cook.sim.database :as db]
+            [cook.sim.schedule :as schedule]
+            [cook.sim.runner :as runner]
+            [cook.sim.reporting :as reporting]
+            [cook.sim.system :as sys]
+            [com.stuartsierra.component :as component]
+            [robert.bruce :refer [try-try-again]]))
+
+
+(defn wait-for-cook
+  [settings]
+  (try-try-again {:sleep 5000 :tries 20
+                  :error-hook #(println "Cook API not up yet:" (.getMessage %))}
+                 ;; will still throw exception on connection refused.
+                 http/get (:cook-api-uri settings) {:throw-exceptions false}))
+
+(defn schedulable?
+  "Returns true iff the job is intended to be schedulable according to the
+  configuration of the Minimesos cluster on travis.  We're accomplishing this
+  by means of a usage profile and and minimesos config that we know to be incompatible
+  - see minimesosFile and simulator_config.clj"
+  [job]
+  (not (string/starts-with? (:username job) "unsched")))
+
+(defn sim-progress
+  [sim-db cook-db sim-id]
+  (let [jobs (reporting/job-results-from-components sim-db cook-db sim-id)
+        [valid-jobs invalid-jobs] ((juxt filter remove) schedulable? jobs)
+        [scheduled-jobs unscheduled-jobs] ((juxt filter remove) :wait-time jobs)
+        [finished-jobs unfinished-jobs] ((juxt filter remove) :turnaround jobs)]
+    {:valid {:total (count valid-jobs)
+             :unscheduled (count (intersection (set valid-jobs) (set unscheduled-jobs)))
+             :unfinished (count (intersection (set valid-jobs) (set unfinished-jobs)))}
+     :invalid {:total (count invalid-jobs)
```

Comment:
Invalid jobs are the ones that intentionally unschedulable.   Should probably rename to unschedulable-jobs since they actually are valid.